### ISSUE TITLE
argo server ui | SSO/Auth Disabled | still trying to authenticate

### ISSUE THREAD

**Issue by mailtoraja18** [argoproj/argo#4428]
I am running oauth2-proxy for authentication and then upstream call to argo ui on a http port. The UI works fine when i directly ingress to the UI with no authentication. But does not work when proxied via oauth proxy after the authentication is successful. The upstream call to APIs fails with 401. 

What happened/what you expected to happen?
Getting 401 unauthorized. Expecting the Argo UI to use the forwarded user headers and login.

What Kubernetes provider are you using? 
EKS
What version of Argo Workflows are you running? 

2.11.3

Logs

$ kubectl -n argo logs -f argo-server-79bf874754-xxzbw argo-server

time="2020-10-31T22:00:35Z" level=debug msg="CLI version" version="{v2.11.3 2020-10-07T22:56:56Z a00a8f141c221f50e397aea8f86a54171441e395 v2.11.3 clean go1.13.4 gc linux/amd64}"
I1031 22:00:35.676852       1 merged_client_builder.go:122] Using in-cluster configuration
I1031 22:00:35.677323       1 merged_client_builder.go:164] Using in-cluster namespace
time="2020-10-31T22:00:35Z" level=info authModes="[server]" baseHRef=/ managedNamespace=argo namespace=argo secure=false
time="2020-10-31T22:00:35Z" level=warning msg="**You are running in insecure mode**. Learn how to enable transport layer security: https://argoproj.github.io/argo/tls/"
time="2020-10-31T22:00:35Z" level=warning msg="You are running without client authentication. Learn how to enable client authentication: https://argoproj.github.io/argo/argo-server-auth-mode/"
time="2020-10-31T22:00:35Z" level=info msg="config map" name=workflow-controller-configmap
time="2020-10-31T22:00:35Z" level=info msg="SSO disabled"
I1031 22:00:36.697927       1 round_trippers.go:443] GET https://10.100.0.1:443/api/v1/namespaces/argo/configmaps/workflow-controller-configmap 200 OK in 1019 milliseconds
time="2020-10-31T22:00:36Z" level=info msg="Starting Argo Server" instanceID= version=v2.11.3
time="2020-10-31T22:00:36Z" level=info msg="Creating event controller" operationQueueSize=16 workerCount=4
time="2020-10-31T22:00:36Z" level=info msg="Argo Server started successfully on http://localhost:2746"
I1031 22:00:36.709152       1 reflector.go:150] Starting reflector *v1.ConfigMap (0s) from pkg/mod/k8s.io/client-go@v0.17.8/tools/cache/reflector.go:105
I1031 22:00:36.709258       1 reflector.go:185] Listing and watching *v1.ConfigMap from pkg/mod/k8s.io/client-go@v0.17.8/tools/cache/reflector.go:105
I1031 22:00:36.710755       1 round_trippers.go:443] GET https://10.100.0.1:443/api/v1/namespaces/argo/configmaps?fieldSelector=metadata.name%3Dworkflow-controller-configmap&limit=500&resourceVersion=0 200 OK in 1 milliseconds
I1031 22:00:36.712775       1 round_trippers.go:443] GET https://10.100.0.1:443/api/v1/namespaces/argo/configmaps?allowWatchBookmarks=true&fieldSelector=metadata.name%3Dworkflow-controller-configmap&resourceVersion=6865664&timeoutSeconds=420&watch=true 200 OK in 1 milliseconds
time="2020-10-31T22:02:19Z" **level=info msg="finished unary call with code Unauthenticated" error="rpc error: code = Unauthenticated desc = no valid authentication methods found for mode client"** grpc.code=Unauthenticated grpc.method=ListWorkflows grpc.service=workflow.WorkflowService grpc.start_time="2020-10-31T22:02:19Z" grpc.time_ms=0.056 span.kind=server system=grpc

Failed request
``` 
Request URL: https://argoweb.maps.dev.aws.aaaaaaaa.net/api/v1/version
Request Method: GET
Status Code: 401 
Remote Address: 100.69.209.147:443
Referrer Policy: no-referrer-when-downgrade
content-type: application/json
date: Sat, 31 Oct 2020 22:02:19 GMT
server: nginx/1.19.1
status: 401
strict-transport-security: max-age=15724800; includeSubDomains
trailer: Grpc-Trailer-Content-Type
:authority: argoweb.maps.dev.aws.aaaaaaaa.net
:method: GET
:path: /api/v1/version
:scheme: https
accept: */*
accept-encoding: gzip, deflate, br
accept-language: en-US,en;q=0.9
cookie: _oauth2_proxy=<redacted>
referer: https://argoweb.maps.dev.aws.aaaaaaaa.net/workflows
sec-fetch-dest: empty
sec-fetch-mode: cors
sec-fetch-site: same-origin
user-agent: Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537
```

**Comment by Amauche** [comment]
Hi, I have exactly the same problem.

**Comment by alexec** [comment]
oauth2-proxy sends you authentication headers in the request. To work-around - reconfigure oauth2-proxy to stop doing this. Fixed in v2.11.7
